mobile/tools/what_to_run.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+set -euo pipefail
+
+BRANCH_NAME="$GITHUB_REF_NAME"
+BASE_COMMIT="$(git merge-base origin/main HEAD)"
+CHANGED_FILES="$(git diff "${BASE_COMMIT}" --name-only)"
+CHANGE_MATCH='^mobile/|^bazel/repository_locations\.bzl|^\.bazelrc|^\.github/workflows/mobile-*|^\.github/workflows/env.yml'
+
+
+run_default_ci () {
+    {
+        echo "mobile_android_build=true"
+        echo "mobile_android_tests=true"
+        echo "mobile_asan=true"
+        echo "mobile_cc_tests=true"
+        echo "mobile_compile_time_options=true"
+        echo "mobile_coverage=true"
+        echo "mobile_formatting=true"
+        echo "mobile_ios_build=true"
+        echo "mobile_ios_tests=true"
+        echo "mobile_release_validation=true"
+        echo "mobile_tsan=true"
+    } >> "$GITHUB_OUTPUT"
+}
+
+run_ci_for_changed_files () {
```

Comment:
can we add some breadcrumb comments while we're in here?  We only run the expensive CI jobs for "high risk" changes: mobile PRs or PRs which touch the files listed in CHANGE_MATCH